platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java

```diff
@@ -210,8 +221,11 @@ private Collection<Project> findProjectsApplyingIdeaAndScalaPlugins() {
         return Collections2.filter(rootProject.getAllprojects(), new Predicate<Project>() {
             @Override
             public boolean apply(Project project) {
-                final boolean hasIdeaPlugin = project.getPlugins().hasPlugin(IdeaPlugin.class);
-                final boolean hasScalaPlugin = project.getPlugins().hasPlugin(ScalaBasePlugin.class);
+                // TODO:isolated Remove unsafe access to other projects' plugin container
+                // This is relatively safe, because we run after projects evaluated, but this might not work for the incremental IP hits
+                PluginContainer pluginContainer = ((ProjectInternal) project).getOwner().getMutableModel().getPluginManager().getPluginContainer();
```

Comment:
💭 Introduce a `Workarounds` (or `WorkaroundsForIP`) class and move that code there? We used that pattern with CC and it worked well (we still have some but they are very easy to find)